catroid/src/main/java/org/catrobat/catroid/io/XstreamSerializer.java

```diff
@@ -585,15 +586,24 @@ public static boolean renameProject(File xmlFile, String dstName) throws IOExcep
 			throw new FileNotFoundException(xmlFile + " does not exist.");
 		}
 
-		String srcName = new ProjectMetaDataParser(xmlFile).getProjectMetaData().getName();
+		String currentXml = Files.toString(xmlFile, Charsets.UTF_8);
+		StringFinder stringFinder = new StringFinder();
+
+		if (!stringFinder.findBetween(currentXml, PROGRAM_NAME_START_TAG, PROGRAM_NAME_END_TAG)) {
+			return false;
+		}
+
+		String srcName = stringFinder.getResult();
+		dstName = new XStream().toXML(dstName);
+		dstName = dstName.replace("<string>", "");
+		dstName = dstName.replace("</string>", "");
```

Comment:
May extract your string operations in an extra function, just to make it a bit more clean to read